iaso/api/tasks/views.py

```diff
@@ -1,92 +1,75 @@
 from rest_framework.decorators import action
 from rest_framework.response import Response
 
-from iaso.models.json_config import Config
-from ..common import ModelViewSet, TimestampField, UserSerializer, HasPermission
 from hat.menupermissions import models as permission
-from iaso.models.base import ERRORED, RUNNING, SKIPPED, KILLED, SUCCESS, QUEUED, Task
+from iaso.api.common import ModelViewSet, HasPermission
+from iaso.api.tasks.filters import (
+    StatusFilterBackend,
+    StartEndDateFilterBackend,
+    TaskTypeFilterBackend,
+    UsersFilterBackend,
+)
+from iaso.models.base import ERRORED, RUNNING, SKIPPED, QUEUED, Task
+from iaso.models.json_config import Config
 from iaso.utils.s3_client import generate_presigned_url_from_s3
 
+from .serializers import ExternalTaskSerializer, ExternalTaskPostSerializer, TaskSerializer
+
 
 task_service = LazyService("BACKGROUND_TASK_SERVICE")
 logger = logging.getLogger(__name__)
 
 
-class TaskSerializer(serializers.ModelSerializer):
-    polio_notification_import_id = serializers.SerializerMethodField()
-
-    class Meta:
-        model = Task
-
-        # Do not include the params, it can contains sensitive information such as passwords
-        fields = [
-            "id",
-            "created_at",
-            "started_at",
-            "ended_at",
-            "progress_value",
-            "end_value",
-            "launcher",
-            "result",
-            "status",
-            "name",
-            "should_be_killed",
-            "progress_message",
-            "polio_notification_import_id",
-        ]
-
-        read_only_fields = ["launcher_name"]
-
-    launcher = UserSerializer(read_only=True)
-    ended_at = TimestampField(read_only=True)
-    created_at = TimestampField(read_only=True)
-    started_at = TimestampField(read_only=True)
-
-    def get_polio_notification_import_id(self, obj: Task) -> Union[int, None]:
-        if obj.name == "create_polio_notifications_async":
-            return obj.params.get("kwargs", {}).get("pk")
-        return None
-
-    def update(self, task, validated_data):
-        if validated_data["should_be_killed"] is not None:
-            task.should_be_killed = validated_data["should_be_killed"]
-            if task.external:
-                task.status = KILLED
-            task.save()
-        return task
-
-
 class TaskSourceViewSet(ModelViewSet):
     """Task API
 
     GET /api/tasks/
     GET /api/tasks/<id>
+    GET /api/tasks/<id>/presigned-url/
+    GET /api/tasks/types/
     PATCH /api/tasks/<id>
+    PATCH /api/tasks/<id>/relaunch/
     """
 
     permission_classes = [
         permissions.IsAuthenticated,
         HasPermission(permission.DATA_TASKS),
     ]  # type: ignore
+    filter_backends = [
+        filters.OrderingFilter,
+        django_filters.rest_framework.DjangoFilterBackend,
+        TaskTypeFilterBackend,
+        UsersFilterBackend,
+        StartEndDateFilterBackend,
+        StatusFilterBackend,
+    ]
```

Comment:
Just for you information (no need to rewrite that) I like to do that via [django-filter](https://django-filter.readthedocs.io/en/latest/guide/rest_framework.html).

It has built-in validation and you don't need a bunch of different classes.

You have [various examples](https://github.com/BLSQ/iaso/blob/5c7e90d95dd93f6cfc1b0f8e4414a3916a552800/plugins/polio/api/chronogram/filters.py#L47) in the code base.

Reply:
Oh I like that, thanks! TIL